src/shared/user-record.c

```diff
@@ -2775,6 +2775,9 @@ int user_record_match(UserRecord *u, const UserDBMatch *match) {
         if (!match)
                 return true;
 
+        if (!uid_is_valid(u->uid))
+                return false;
```

Comment:
So, hmm, thinking more about it, I'd probably move this check before !match, and return -EINVAL instead. The comparison is utterly broken anyway if the input uid is invalid. Plus the function is allowed to fail

Reply:
That's a fair point, I didn't notice the return value (and most callers) use `int`.
Let me pass the error, fix callers and rely on tests...